source/Patches/Roles/Lover.cs

```diff
@@ -71,22 +71,22 @@ public static void Gen(List<PlayerControl> crewmates, List<PlayerControl> impost
         {
             //System.Console.WriteLine("LOVER2");
             if (crewmates.Count <= 0) return;
-            if (crewmates.Count <= 1 && impostors.Count < 1) return;
+            if (crewmates.Count <= 1 && (impostors.Count < 1 || !CustomGameOptions.AllowLovingImpostor)) return;
 
             //System.Console.WriteLine("LOVER3");
             var b = Random.RandomRangeInt(0, 3);
```

Comment:
slightly out of scope i know, but could `b` be changed to just be the `lovingImpostor` boolean?

seems it was originally just meant to be `1` or `0` indicating if a loving impostor should be generated

Reply:
I wanted to, but it urns out at 1am after like 5 hours of Among Us, my brain couldn't figure out how. I've attempted to rewrite this whole section to not reassign stuff to the same variable.